### ISSUE TITLE
Loading the replay mod gives a blank screen.

### ISSUE THREAD

**Issue by thatdude69420** [unhappychoice/ReplayMod#1]
Hi Mr LeoCraft,

I am on 1.16.5 and the replay mod is working whilst I am in-game, as it is recording and configurable. It is even showing up in the replay folder on the main menu. But when I go to load the actual replay file, the screen does not display anything. It just the blank minecraft background. (like when you are loading into a world; that's the background.) The game is not crashing. It's just staying on the blank background; so I would have to alt-tab to end the minecraft. Any tips?

I am not knowledgeable on programming or anything.

**Comment by unhappychoice** [comment]
please upload the screenshot and logs/latest.log file!

**Comment by thatdude69420** [comment]
![Screenshot_1](https://user-images.githubusercontent.com/83432673/116635204-276c5b80-a92c-11eb-96d9-8e11b63b3864.png)
![Screenshot_2](https://user-images.githubusercontent.com/83432673/116635206-276c5b80-a92c-11eb-94d7-b23ea6235662.png)

[latest.log](https://github.com/unhappychoice/ReplayMod/files/6402420/latest.log)

**Comment by thatdude69420** [comment]
Good day Mr unhappychoice,

So I tried using your valhesia's structure jar file; but it ends up crashing the minecraft (I even installed the valhesia's core mod as well). So I decided to completely remove both vahesia's structure and the core. However, when I load the replay file it does show that its loading terrain, but its just stuck on that. I am hearing ambient sounds and music whilst I am in this loading screen. 
Also, at the top left corner of the screenshot (not too sure why it isn't showing up in the screenshot) it says: Minecraft 1.16.5 Multiplayer (3rd party server) when I am only playing singleplayer. I don't know if that's intended? 

![Screenshot_2](https://user-images.githubusercontent.com/83432673/116767239-0293e900-a9fd-11eb-9b21-2c95f9afef0d.png)

[latest.log](https://github.com/unhappychoice/ReplayMod/files/6408721/latest.log)

TLDR: So the changes are: I am seeing the words "loading terrain" versus it being completely blank last time. And I tried placing your valhesia's structure jar file in the mod folder (then needed the core mod) but ended up crashing the minecraft.

Finally, you indicated to build the jar file. But I don't know how to. I don't know much programming. Can you elaborate a little more? I am very doubtful that an updated valhesia's structure version would drop anytime soon.

Your willingness to help is greatly appreciated. I eagerly await some good news.

**Comment by thatdude69420** [comment]
Good day Mr unhappychoice,

So I tried using your valhesia's structure jar file; but it ends up crashing the minecraft (I even installed the valhesia's core mod as well). So I decided to completely remove both vahesia's structure and the core. However, when I load the replay file it does show that its loading terrain, but its just stuck on that. I am hearing ambient sounds and music whilst I am in this loading screen. 
Also, at the top left corner of the screenshot (not too sure why it isn't showing up in the screenshot) it says: Minecraft 1.16.5 Multiplayer (3rd party server) when I am only playing singleplayer. I don't know if that's intended? 

![Screenshot_2](https://user-images.githubusercontent.com/83432673/116767239-0293e900-a9fd-11eb-9b21-2c95f9afef0d.png)

[latest.log](https://github.com/unhappychoice/ReplayMod/files/6408721/latest.log)

TLDR: So the changes are: I am seeing the words "loading terrain" versus it being completely blank last time. And I tried placing your valhesia's structure jar file in the mod folder (then needed the core mod) but ended up crashing the minecraft.

Finally, you indicated to build the jar file. But I don't know how to. I don't know much programming. Can you elaborate a little more? I am very doubtful that an updated valhesia's structure version would drop anytime soon.

Your willingness to help is greatly appreciated. I eagerly await some good news.

**Comment by unhappychoice** [comment]
https://github.com/mezz/JustEnoughItems/issues/2233

**Comment by unhappychoice** [comment]
You can find JEI causes another problem in the latest.log.
I need more investigation to decide whether ReplayMod should handle this or not.

But you can try other JEI versions for temporary fix ( I think a little old one is better. 


```
java.lang.IllegalStateException: A JEI API method is being called by another mod from the wrong thread:
Thread[replaymod-async-sender,10,main]
It must be called on the main thread by using Minecraft.addScheduledTask.
	at mezz.jei.util.ErrorUtil.assertMainThread(ErrorUtil.java:257) ~[jei:7.6.4.87]
	at mezz.jei.recipes.RecipeManager.unhideRecipe(RecipeManager.java:308) ~[jei:7.6.4.87]
	at mezz.jei.recipes.RecipeManager.addRecipe(RecipeManager.java:148) ~[jei:7.6.4.87]
	at mezz.jei.recipes.RecipeManager.addRecipeTyped(RecipeManager.java:105) ~[jei:7.6.4.87]
	at mezz.jei.recipes.RecipeManager.addRecipes(RecipeManager.java:98) ~[jei:7.6.4.87]
	at mezz.jei.recipes.RecipeManager.<init>(RecipeManager.java:90) ~[jei:7.6.4.87]
	at mezz.jei.load.PluginLoader.getRecipeManager(PluginLoader.java:142) ~[jei:7.6.4.87]
	at mezz.jei.startup.JeiStarter.start(JeiStarter.java:74) ~[jei:7.6.4.87]
	at mezz.jei.startup.ClientLifecycleHandler.startJEI(ClientLifecycleHandler.java:141) ~[jei:7.6.4.87]
	at mezz.jei.startup.ClientLifecycleHandler.setupJEI(ClientLifecycleHandler.java:134) ~[jei:7.6.4.87]
	at mezz.jei.startup.ClientLifecycleHandler.lambda$new$2(ClientLifecycleHandler.java:100) ~[jei:7.6.4.87]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) [eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) [eventbus-4.0.0.jar:?]
	at net.minecraft.tags.ITagCollectionSupplier.func_242212_e(ITagCollectionSupplier.java:25) [?:?]
	at net.minecraft.client.network.play.ClientPlayNetHandler.handler$zef000$replayMod_ignoreHandshakeConnectionClose(ClientPlayNetHandler.java:2784) [?:?]
	at net.minecraft.client.network.play.ClientPlayNetHandler.func_199723_a(ClientPlayNetHandler.java) [?:?]
	at net.minecraft.network.play.server.STagsListPacket.func_148833_a(SourceFile:31) [?:?]
	at net.minecraft.network.play.server.STagsListPacket.func_148833_a(SourceFile:9) [?:?]
	at net.minecraft.network.NetworkManager.func_197664_a(NetworkManager.java:148) [?:?]
	at net.minecraft.network.NetworkManager.channelRead0(NetworkManager.java:138) [?:?]
	at net.minecraft.network.NetworkManager.channelRead0(NetworkManager.java:49) [?:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105) [netty-all-4.1.25.Final.jar:4.1.25.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) [netty-all-4.1.25.Final.jar:4.1.25.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) [netty-all-4.1.25.Final.jar:4.1.25.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) [netty-all-4.1.25.Final.jar:4.1.25.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) [netty-all-4.1.25.Final.jar:4.1.25.Final]
	at com.replaymod.replay.FullReplaySender.channelRead(FullReplaySender.java:398) [replaymod:?]
	at com.replaymod.replay.FullReplaySender$2.run(FullReplaySender.java:946) [replaymod:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_291]
```

**Comment by thatdude69420** [comment]
Hey unhappychoice,

So I removed jei and it actually works. I am surprised!! Its strange because even though I removed jei, the jei hud shows up in game still. Either way thank you so much for the assistance. But let me ask you something. Once I add more mods to my minecraft; there is a chance with each mod, to break the replay mod again right? So I'll always have to be checking the log file to see which mod is breaking it?

I presume that you developed this mod to shoot your youtube videos right. And once it works for you, you'll see no reason to update to fix any bugs right? If so, can you recommend like a youtube video of the sort so I'll be able to fix issues without bothering you?

And once again, thank you.

**Comment by unhappychoice** [comment]
> Its strange because even though I removed jei, the jei hud shows up in game still

That's because some mods include JEI dependency ( It's not accurate / correct way, I think

> Once I add more mods to my minecraft; there is a chance with each mod, to break the replay mod again right?

Yes. 
Right now, ReplayMod emulates vanilla server to replay recorded data even it's not vanilla environment.
And I think most of mods don't assume that, of course, then crash, make a bug, freeze, etc.

>  So I'll always have to be checking the log file to see which mod is breaking it?

Yes.
To prevent that, we need do something like

- Rewrite whole structure / how this works
- Handle crash or bugs made by every other mods 

But these need much time and cost.
So, I need to be careful to decide something about that.

> I presume that you developed this mod to shoot your youtube videos right

That's true.

 > And once it works for you, you'll see no reason to update to fix any bugs right?
 
 Not any. If it's easy or critical, I will fix that.
 
 > If so, can you recommend like a youtube video of the sort so I'll be able to fix issues without bothering you?
 
 Yes, I can.
 But fixing the problems by coding is better in most cases, so it depends
 
 > And once again, thank you.

**Comment by thatdude69420** [comment]
Thank you for your reply. I wish you continued success.

**Comment by thatdude69420** [comment]
Hi Mr unhappychoice,

Can you look at this log file and tell which mod is conflicting with epic fight mod? I really wanna keep the epic fight mod but I don't know which mode to delete. I am having no problems when epic fight mod is not used.

I am not getting any replies on the official mod page and you're the only one I know that can read this. Please?

I removed the replaymod to see if it'll work but still.

[latest.log](https://github.com/unhappychoice/ReplayMod/files/6467562/latest.log)

**Comment by unhappychoice** [comment]
https://github.com/Yesssssman/epicfightmod/issues/245
It seems 👆and it's purely the bug of epic fight.
So, there's nothing ReplayMod can do.

**Comment by thatdude69420** [comment]
thanks leo. great videos btw! will continue to support. my name is body mind zone on youtube.
